File: 프로그래머스/1/42840. 모의고사/모의고사.js
function solution(answers) {
    const answer = [];
    const result = [0, 0, 0];
    
    const arr = [[1, 2, 3, 4, 5], [2, 1, 2, 3, 2, 4, 2, 5], [3, 3, 1, 1, 2, 2, 4, 4, 5, 5]]
    
    arr.forEach((line, index) => {
        let score = 0;
        for (let i = 0; i < answers.length; i++) {
            if (answers[i] === line[i % line.length]) result[index] += 1 
        }
    })
    
    let max = Math.max(...result)
    
    for (let i = 0; i < result.length; i++) {
        if (result[i] === max) answer.push(i + 1)
    }
    
    
    return answer;
}

File: 프로그래머스/1/42840. 모의고사/모의고사.py
def solution(answers):
    patterns = [
        [1,2,3,4,5],
        [2,1,2,3,2,4,2,5],
        [3,3,1,1,2,2,4,4,5,5]
    ]
    
    score = [0] * 3
    
    for i, answer in enumerate(answers):
        for j, pattern in enumerate(patterns):
            if answer == pattern[i % len(pattern)]:
                score[j] += 1
                
    max_score = max(score)
    
    answer = []
    for i in range(3):
        if score[i] == max_score:
            answer.append(i + 1)
    
    return answer